### PR TITLE
RMMIS-5122: remove form code, make state updates use Immutable struct…

### PR DESCRIPTION
**Pull request by mallenscdhhs**
…ures, fix issue where edits where not persisted

### CODE REVIEW

**Comment by kjschulz** [comment]
Hmm, tests are failing and it won't run for me locally for some reason.

File: src/EntryList.jsx [hunk 1]
```diff
@@ -60,17 +51,11 @@ class EntryList extends React.Component {
     // when the user clicks the edit entry link
```
Comment (kjschulz) [inline src/EntryList.jsx]: On line 36, you no longer need `this.renderForm = this.renderForm.bind(this);` it's causing some pretty wacky Dispatcher errors

File: src/EntryListForm.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  actionName: React.PropTypes.string,
+  actionEvent: React.PropTypes.string
+};
+
+EntryListForm.defaultProps = {
+  show: false,
+  config: {},
+  actionName: '',
+  actionEvent: constants.actions.ENTRYLIST_NEW_ENTRY_ADD
+}
```
Comment (kjschulz) [inline src/EntryListForm.jsx]: Missing a semicolon here.

**Comment by mallenscdhhs** [comment]
Man, the stomach bug I had must've eaten my brain too. Will fix.

File: src/EntryList.jsx (hunk 1 above)
Comment (mallenscdhhs) [inline src/EntryList.jsx]: fixed :facepunch:

Comment (kjschulz) [inline src/EntryList.jsx]: WOOT!  :facepunch:

**Comment by kjschulz** [comment]
:+1:

**Pull request merged**